web/app/js/components/Sidebar.jsx

```diff
@@ -104,29 +108,40 @@ class Sidebar extends React.Component {
     });
   }
 
-  toggleCollapse() {
-    if (this.state.initialCollapse) {
-      // fix weird situation where toggleCollapsed is called on pageload,
-      // causing the toggle states to be inconsistent. Don't toggle on the
-      // very first call to toggleCollapse()
-      this.setState({ initialCollapse: false});
-    } else {
-      this.setState({ collapsed: !this.state.collapsed });
+  handleNamespaceSelector(value) {
+    this.setState({namespaceFilter: value});
+  }
+
+  // Filters resources retrieved from a stat query by namespaces. Does not include resources of type "authority"
+  filterResourcesByNamespace(resources, namespace) {
+    if (namespace === "all") {
+      return _.omit(resources, ["authority"]);
     }
+
+    let result = _.mapValues(resources, o => {
+      return _(o)
+        .filter(r => r.resource.namespace === namespace)
+        .value();
+    });
+    return _.omit(result, ["authority"]);
+  }
+
+  generateResourceURL(r) {
+    return "/namespaces/" + r.resource.namespace + "/" + r.resource.type + "s/" + r.resource.name;
   }
 
   render() {
     let normalizedPath = this.props.location.pathname.replace(this.props.pathPrefix, "");
     let PrefixedLink = this.api.PrefixedLink;
-    let numHiddenNamespaces = _.size(this.state.namespaces) - this.state.maxNsItemsToShow;
+    let namespaces = [{nsValue: "all", nsName: "All Namespaces"}].concat(_.map(this.state.namespaces, ns => {
+      return {nsValue: ns, nsName: ns};
+    }));
+    let sidebarComponents = this.filterResourcesByNamespace(this.state.resourceGroupings, this.state.namespaceFilter);
 
     return (
       <Layout.Sider
         width="260px"
-        breakpoint="lg"
-        collapsed={this.state.collapsed}
```

Comment:
Per discussion on the ticket, I think we'll want to retain the ability to collapse (or hide) the sidebar.